pkg/client/unversioned/restclient.go

```diff
@@ -88,6 +89,9 @@ func (c *RESTClient) Verb(verb string) *Request {
 	if c.Throttle != nil {
 		c.Throttle.Accept()
 	}
+	if c.Client == nil {
+		return NewRequest(nil, verb, c.baseURL, c.apiVersion, c.Codec)
+	}
```

Comment:
`NewRequest` takes an interface as its first parameter. You can think of an interface as a tuple of value and the type of the value. `(v, t)`. A nil check against and interface(`intf == nil`) is only true if both `t` AND `v` are nil. `c.Client` is a nil value, but it has a type, `*http.Client`. So when we pass it, it is `(nil, *http.Client)`. The nil check in the request code doesn't catch this, tries to use it, then panics. To prevent this, we pass an untyped nil here so it gets `(nil, nil)` and the nil check in the request code works.

http://research.swtch.com/interfaces